README.md

```diff
@@ -190,11 +190,487 @@ autotools-snap --files ./components/**/*.meta.ts
 
 - `skip-on-missing-key`: set this flag if you want to skip testing when `process.env.EYES_API_KEY` or `process.env.APPLITOOLS_API_KEY` variables are not set. By default, snap will fail if either of these keys are not set. Example usage: `snap --skip-on-missing-key`, or, `snap -s`.
 
+#### Snap Configuration
+
+Configuration is as simple as adding a `autotools.config.js` file in your `.autotools` folder.
+If you wish to use a configuration file named something different, or located somewhere else, you can specify a specific file using the `--config` flag, like so: 
+
+`yarn snap --config ./config/myConfig.js`
+
+The format of the `autotools.config.js` file is as follows:
+
+``` js
+module.exports = {
+    applitoolsApiKeyName: ‘WHATEVER_YOU_NAMED_YOUR_ENV_VARIABLE’,
+    projectHooks: [],
+    componentHooks: [],
+    simHooks: [],
+    afterHooks: []
+}
+```
+All fields are optional. However, without hooks, Snap won't do anything.
+
+#### Building
+
+Each project may have a different or unique build step. Components may be Stylable-based, or use Sass, or css-modules, or have special imports, or be written in Typescript... the list goes on. In order to keep Snap generic, and usable in many different situations, we've implemented what we're calling *hooks*.
+
+Hooks allow a project to build itself, and then give Snap the information it needs in order to provide screenshot regression tests.
+
+This is our contract, in its simplest terms: Snap will call all of the hooks in the config in a specific order, passing information about the project, and after all the hooks have been called, Snap expects to receive a list containing the locations of HTML files which need to be sent to Applitools. Luckily, we already provide some hooks that cover common-use cases, so as a project, you may not need to write your own.
```

Comment:
We may be missing basic background about how snap works - from simulation to HTML/CSS to screenshot test